flink-clients/src/main/java/org/apache/flink/client/program/StreamContextEnvironment.java

```diff
@@ -269,20 +274,86 @@ private List<String> collectNotAllowedConfigurations() {
                                         ConfigurationNotAllowedMessage.ofConfigurationChange(
                                                 k, v)));
 
-        if (!Arrays.equals(originalCheckpointConfigSerialized, serializeConfig(checkpointCfg))) {
+        final Configuration enrichedClusterConfig = new Configuration(originalClusterConfig);
+        enrichProgramConfigWildcards(enrichedClusterConfig);
+
+        // Check CheckpointConfig
+        final CheckpointConfig clusterCheckpointConfig = new CheckpointConfig();
+        clusterCheckpointConfig.configure(enrichedClusterConfig);
```

Comment:
can we simply use the `configuration` here? (we've already checked whether it only contains allowed overrides)

Reply:
No we can't. Configuration is spread across `configuration` and the POJOs, so we need to have logic for comparing both. The first by removing keys, the second by setting fields before comparison.